module/zfs/arc.c

```diff
@@ -6268,14 +6269,22 @@ arc_read(zio_t *pio, spa_t *spa, const blkptr_t *bp,
 			if (vd != NULL)
 				spa_config_exit(spa, SCL_L2ARC, vd);
 			/*
-			 * Skip ARC stat bump for block pointers with
-			 * embedded data. The data are read from the blkptr
-			 * itself via decode_embedded_bp_compressed().
+			 * If it wasn't even supposed to be in L2, that's
+			 * not a miss
 			 */
-			if (l2arc_ndev != 0 && !embedded_bp) {
-				DTRACE_PROBE1(l2arc__miss,
-				    arc_buf_hdr_t *, hdr);
-				ARCSTAT_BUMP(arcstat_l2_misses);
+			if (l2arc_ndev != 0 && HDR_L2CACHE(hdr)) {
+				/*
+				 * Skip ARC stat bump for block pointers
+				 * with embedded data. The data are read from
+				 * the blkptr itself via
+				 * decode_embedded_bp_compressed().
+				 */
+				if (!embedded_bp) {
+					DTRACE_PROBE1(l2arc__miss,
+								    arc_buf_hdr_t *,
+								    hdr);
```

Comment:
You don't have to break a new line here. There should be space in the previous one.

Reply:
checkstyle complained that - since I'm indenting one extra level - that line is now >80 chars :(